server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -3883,6 +3893,14 @@ private Tuple<BlobStoreIndexShardSnapshots, ShardGeneration> buildBlobStoreIndex
                     generation
                 );
             } catch (NoSuchFileException noSuchFileException) {
+                final long latestGeneration = latestIndexBlobId();
+                final RepositoryData currentRepositoryData = getRepositoryData(latestGeneration);
```

Comment:
These lines can also fail with an exception - if they do, I'd rather we suppressed the exception they throw and went back to throwing the original `noSuchFileException`.

Reply:
Good point as well. See fe1cf86524225038ef1b9ea31ce0c2b227223c87